Mathlib/Geometry/Manifold/Orientable.lean

```diff
@@ -0,0 +1,192 @@
+  simp [ne_of_gt FiniteDimensional.finrank_pos]
+
+lemma orientationPreserving_id (s : Set H) : OrientationPreserving id s := by
+  intro
+  simp [ContinuousLinearMap.det]
+
+lemma orientationPreserving_comp [FiniteDimensional ℝ H] {f g : H → H} {u v : Set H}
+    (hf : OrientationPreserving f u) (hg : OrientationPreserving g v) :
+    OrientationPreserving (g ∘ f) (u ∩ f ⁻¹' v) := by
+  intro x ⟨hxu, hxv⟩
+  rw [fderiv.comp x (hg.differentiableAt hxv) (hf.differentiableAt hxu)]
+  unfold ContinuousLinearMap.det ContinuousLinearMap.comp
+  rw [(fderiv ℝ g (f x)).toLinearMap.det_comp (fderiv ℝ f x).toLinearMap]
+  exact mul_pos (hg (f x) hxv) (hf x hxu)
+
+lemma orientationReversing_comp_orientationPreserving [FiniteDimensional ℝ H]
+    {f g : H → H} {u v : Set H} (hf : OrientationPreserving f u) (hg : OrientationReversing g v) :
+    OrientationReversing (g ∘ f) (u ∩ f ⁻¹' v) := by
+  intro x ⟨hxu, hxv⟩
+  rw [fderiv.comp x (hg.differentiableAt hxv) (hf.differentiableAt hxu)]
+  unfold ContinuousLinearMap.det ContinuousLinearMap.comp
+  rw [(fderiv ℝ g (f x)).toLinearMap.det_comp (fderiv ℝ f x).toLinearMap]
+  exact mul_neg_of_neg_of_pos (hg (f x) hxv) (hf x hxu)
+
+lemma orientationPreserving_comp_orientationReversing [FiniteDimensional ℝ H]
+    {f g : H → H} {u v : Set H} (hf : OrientationReversing f u) (hg : OrientationPreserving g v) :
+    OrientationReversing (g ∘ f) (u ∩ f ⁻¹' v) := by
+  intro x ⟨hxu, hxv⟩
+  rw [fderiv.comp x (hg.differentiableAt hxv) (hf.differentiableAt hxu)]
+  unfold ContinuousLinearMap.det ContinuousLinearMap.comp
+  rw [(fderiv ℝ g (f x)).toLinearMap.det_comp (fderiv ℝ f x).toLinearMap]
+  exact mul_neg_of_pos_of_neg (hg (f x) hxv) (hf x hxu)
+
+lemma orientationReversing_comp {f g : H → H} {u v : Set H}
+    (hf : OrientationReversing f u) (hg : OrientationReversing g v) :
+    OrientationPreserving (g ∘ f) (u ∩ f ⁻¹' v) := by
+  intro x ⟨hxu, hxv⟩
+  rw [fderiv.comp x (hg.differentiableAt hxv) (hf.differentiableAt hxu)]
+  unfold ContinuousLinearMap.det ContinuousLinearMap.comp
+  rw [(fderiv ℝ g (f x)).toLinearMap.det_comp (fderiv ℝ f x).toLinearMap]
+  exact mul_pos_of_neg_of_neg (hg (f x) hxv) (hf x hxu)
+
+/-- The pregroupoid of orientation-preserving maps. -/
+def orientationPreservingPregroupoid [FiniteDimensional ℝ H] : Pregroupoid H where
+  property f s := OrientationPreserving f s
+  comp hf hg _ _ _ _ hx := orientationPreserving_comp hf hg _ hx
+  id_mem := orientationPreserving_id _
+  locality {f u} _ h x hxu :=
+    have ⟨v, _, hxv, h⟩ := h x hxu
+    h x <| Set.mem_inter hxu hxv
+  congr {f g u} hu fg hf x hx := by
+    rw [(Filter.eventuallyEq_of_mem (hu.mem_nhds hx) fg).fderiv_eq]
+    exact hf x hx
+
+/-- The groupoid of orientation-preserving maps. -/
+def orientationPreservingGroupoid [FiniteDimensional ℝ H] : StructureGroupoid H :=
+  orientationPreservingPregroupoid.groupoid
+
+end OrientationPreserving
+
+section OrientableManifold
```

Comment:
I personally prefer the comment on top of the section name. I wonder if the current behaviour is also incidental due to the port. Not a strong opinion, in any way.
```
 /-! ### Orientable manifolds -/

section OrientableManifold
```